### PR TITLE
#139284011 Make the API Endpoints RESTful

### PR DESCRIPTION
**Pull request by jackiewmacharia**
#### What does this PR do?
Enables RESTful API endpoints for: Users, Shops, Products and Orders

 #### Description of Task to be completed?
The Project currently uses DDP protocol to communicate between the client and the server. Provide a RESTful API package to allow this communication to also happen over RESTful services. This would allow alternate/external frameworks and systems, that either do not work with Meteor.js or require a more traditional endpoint, to interact with Reaction Commerce.

 #### How should this be manually tested?
* Clone this branch
* Run `meteor npm install`
* Run `reaction` to run the app

###### Using GraphiQL:

* Navigate to `http://localhost:9090/graphql`
* Make the different queries

**Queries**

```
{
  users {
    verified
    fullName
    emails
    verified
    createdAt
    userId
    shopId
  }
  products {
    title
    _id
    vendor
    createdAt
    price
    inventoryQuantity
	}
  shops {
    name
    _id
    status
    currency
    emails
    lastUpdated
  }
  orders (emailID: "admin") {
    deliveryAddress {
      fullName
      streetAddress
      city
      region
      country
    }
    email
    _id
    items {
      title
      quantity
      price
    }
    userId
    workflowStatus
    sessionId
    shopId
    orderDate
    shipped
    tracking
  }
}
```

###### Using Postman:

Make the following get requests on Postman:

>http://localhost:9090/api/products/
>http://localhost:9090/api/users/
>http://localhost:9090/api/shops/
>http://localhost:9090/api/ordered_items/admin
>http://localhost:9090/api/orders_processed/admin
>http://localhost:9090/api/orders_cancelled/admin

 #### What are the relevant pivotal tracker stories?
Feature #139284011

 #### Screenshots (if appropriate)

_**Users**_
users

_**Shops**_
shops

_**Products**_
products

_**Orders**_
orders

### CODE REVIEW

File: server/api/graphql/graphqlSchema.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  name: "Products",
+  description: "Lists select Product fields",
+  fields: () => ({
+    title: { type: GraphQLString },
+    _id: { type: GraphQLString },
+    vendor: { type: GraphQLString },
+    createdAt: {
+      type: GraphQLString,
+      resolve: (obj) => {
+        return obj.createdAt;
```
Comment (warenga) [inline server/api/graphql/graphqlSchema.js]: @andela-jmacharia any reason why we need to specify the resolver when the field `createdAt` matches the `createdAt` property of the response object? From my understanding you don't need to specify if the names are the same, the default resolver will do that for you.

File: server/api/graphql/graphqlSchema.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+          return obj.emails[0].verified;
+        } else {
+          return null;
+        }
+      }
+    },
+    createdAt: {
+      type: GraphQLString,
+      resolve: (obj) => {
+        return obj.createdAt;
```
Comment (warenga) [inline server/api/graphql/graphqlSchema.js]: @andela-jmacharia refer to above comment on the same.

File: server/api/graphql/graphqlSchema.js (hunk 1 above)
Comment (jackiewmacharia) [reply server/api/graphql/graphqlSchema.js]: @andela-wmaina you are right, using the resolve is redundancy in cases where the field name is `createdAt`. Rectified and changes pushed.

File: server/api/graphql/graphqlSchema.js (hunk 2 above)
Comment (jackiewmacharia) [reply server/api/graphql/graphqlSchema.js]: @andela-wmaina done

**Review by warenga — APPROVED** [review]

**Review by jkanyua — APPROVED** [review]

**Pull request merged**